include/boost/geometry/algorithms/detail/overlay/clip_linestring.hpp

```diff
@@ -86,9 +86,10 @@ class liang_barsky
     inline bool clip_segment(Box const& b, segment_type& s, bool& sp1_clipped, bool& sp2_clipped) const
     {
         typedef typename select_coordinate_type<Box, Point>::type coordinate_type;
+        typedef double calc_type;
```

Comment:
Do here type `double` is always used as an intermediate calculation type. This may be a problem in the case of user-defined coordinate types which could have greater precision than `double`, in such a case the precision would be lost during casting done above. In such cases we use `select_most_precise<>` utility (see the `geometry/util` directory). Here it could be used like this:

    typedef typename select_most_precise<coordinate_type, double>::type calc_type;